daft/catalog/__glue.py

```diff
@@ -0,0 +1,586 @@
+        gc.glue = catalog._client
+
+        # the pyiceberg table will hold a ref to gc
+        return gc._convert_glue_to_iceberg(table)
+
+    def read(self, **options) -> DataFrame:
+        from daft.io._iceberg import read_iceberg
+
+        return read_iceberg(
+            table=self._pyiceberg_table, snapshot_id=options.get("snapshot_id"), io_config=self._io_config
+        )
+
+    def write(self, df: DataFrame, mode: Literal["append", "overwrite"] = "append", **options) -> None:
+        df.write_iceberg(self._pyiceberg_table, mode=mode)
+
+
+class GlueDeltaTable(GlueTable):
+    _io_config: IOConfig | None
+    _unity_catalog_table: UnityCatalogTable
+
+    def __init__(self):
+        raise ValueError("GlueIcebergTable.__init__() not supported!")
+
+    @classmethod
+    def from_table_info(cls, catalog: GlueCatalog, table: GlueTableInfo) -> GlueTable:
+        parameters: Parameters = table.get("Parameters", {})
+
+        # verify we have table_type = "delta"
+        table_type = parameters.get("table_type")
+        if table_type is None:
+            raise ValueError("GlueDeltaTable is missing the required 'table_type' parameter.")
+        if table_type.lower() != "delta":
+            raise ValueError(f"GlueDeltaTable had table_type {table_type}, but expected 'delta'")
+
+        t = GlueDeltaTable.__new__(GlueDeltaTable)
+        t._catalog = catalog
+        t._table = table
+        t._io_config = None  # todo
+        t._unity_catalog_table = cls._create_unity_catalog_table(table, None)
+
+        return t
+
+    @classmethod
+    def _create_unity_catalog_table(cls, table: GlueTableInfo, io_config: IOConfig | None) -> UnityCatalogTable:
+        from unitycatalog.types import TableInfo
+
+        from daft.unity_catalog import UnityCatalogTable
+
+        table_info = TableInfo(
+            # catalog_name=
+            # columns=[],
+            comment=None,
+            created_at=int(table["CreateTime"].timestamp() * 1000),
+            data_source_format="DELTA",
+            schema_name=table["DatabaseName"],
+        )
+
+        table_uri = table["StorageDescriptor"]["Location"]
+
+        return UnityCatalogTable(table_info, table_uri, io_config)
+
+    def read(self, **options) -> DataFrame:
+        from daft.io._deltalake import read_deltalake
+
+        return read_deltalake(
+            table=self._unity_catalog_table,
+            version=options.get("version"),
+            io_config=self._io_config,
+        )
+
+    def write(self, df: DataFrame, mode: Literal["append", "overwrite"] = "append", **options) -> None:
+        raise NotImplementedError
+
+
+def _convert_glue_schema(columns: list[GlueColumnInfo]) -> Schema:
+    return Schema._from_fields([_convert_glue_column(column) for column in columns])
+
+
+def _convert_glue_column(column: GlueColumnInfo) -> Field:
+    return Field.create(column["Name"], _convert_glue_type(column["Type"]))
+
+
+def _convert_glue_type(type: str) -> DataType:
```

Comment:
Btw what spec are we using for the types? Could we add a link here? I assumed it would be Hive types but not 100% on this

Reply:
Here's a reference we can use, https://docs.aws.amazon.com/glue/latest/dg/glue-types.html#glue-types-catalog

> This type system is based on Apache Hive's type system. For more information about the Apache Hive type system, see [Types](https://cwiki.apache.org/confluence/display/hive/languagemanual+types) in the Apache Hive wiki.

I'll update the _convert_glue_type to be better.